controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -1154,6 +1165,26 @@ func addressChanged(oldAddress Address, newAddress Address) bool {
 		return true
 	}
 
+	// If the zone hints have changed, then the address has changed
+	if len(newAddress.ForZones) != len(oldAddress.ForZones) {
+		return true
+	}
+
+	// Sort the zone information so that we can compare them accurately
+	sort.Slice(oldAddress.ForZones, func(i, j int) bool {
+		return oldAddress.ForZones[i].Name < (oldAddress.ForZones[j].Name)
+	})
```

Comment:
I took a look into this and it's not possible for this object. It's because it's an array of structs instead of an array of strings.  In theory, we could add the Sort interface to `ForZone` struct, but I'm really hesitant to start adding interfaces to struct I don't control

Reply:
Thanks for the explanation. Makes sense. We can leave it like this, it's not hard to reason with what's happening and it's definitely better than implementing an interface on it or overcomplicating the sort.